core/modules/file/file.module

```diff
@@ -4,9 +4,100 @@
  * Defines a "managed_file" Form API field and a "file" field for Field module.
  */
 
+/**
+ * Modules should return this value from hook_file_entity_access() to allow
+ * access to a file.
+ */
+define('FILE_ACCESS_ALLOW', 'allow');
+
+/**
+ * Modules should return this value from hook_file_entity_access() to deny
+ * access to a file.
+ */
+define('FILE_ACCESS_DENY', 'deny');
+
+/**
+ * Modules should return this value from hook_file_entity_access() to not affect
+ * file access.
+ */
+define('FILE_ACCESS_IGNORE', NULL);
+
+/**
+ * The {file_managed}.type value when the file type has not yet been determined.
+ */
+define('FILE_TYPE_NONE', 'undefined');
+
+/**
+ * File types constant for user-defined file types.
+ */
+define('FILE_TYPE_STORAGE_NORMAL', 1);
+
+/**
+ * File type constant for file types that override module-defined presets.
+ */
+define('FILE_TYPE_STORAGE_OVERRIDE', 2);
+
+/**
+ * File type constant for module-defined file types.
+ */
+define('FILE_TYPE_STORAGE_DEFAULT', 4);
+
 // Load all Field module hooks for File.
 require_once BACKDROP_ROOT . '/core/modules/file/file.field.inc';
 
+/**
+ * Implements hook_hook_info().
+ */
+function file_hook_info() {
+  $hooks = array(
+    'file_operations',
+    'file_type_info',
+    'file_type_info_alter',
+    'file_formatter_info',
+    'file_formatter_info_alter',
+    'file_view',
+    'file_view_alter',
+    'file_displays_alter',
+    'file_type',
+    'file_type_alter',
+    'file_download_headers_alter',
+    'file_entity_access',
+  );
+
+  return array_fill_keys($hooks, array('group' => 'file'));
+}
+
+/**
+ * Implements hook_hook_info_alter().
+ *
+ * Add support for existing core hooks to be located in modulename.file.inc.
+ */
+function file_hook_info_alter(&$info) {
+  $hooks = array(
```

Comment:
Seems like this could be combined into `file_hook_info()` rather than using an alter, now that this is all owned by File module directly.